### ISSUE TITLE
XHR restriction warning for file:// URIs

### ISSUE THREAD

**Issue by FND** [alex-seville/blanket#175]
TLDR: Blanket should catch XHR exceptions when run off a `file://` URI, informing the user of alternatives

While Blanket does an excellent job of integrating with QUnit, my first encounter was somewhat baffling due to XHR restrictions.

My repositories generally look like this:

```
.
├── src
│   ├── foo.js
│   └── bar.js
└── test
    ├── index.html
    ├── test_foo.js
    └── test_bar.js
```

Thus the test suite (`index.html`) references source files via relative paths to the parent directory (`../src/foo.js`).

However, this leads to errors like "NS_ERROR_DOM_BAD_URI: Access to restricted URI denied" or "XMLHttpRequest cannot load file:///.../foo.js. Cross origin requests are only supported for HTTP." when Blanket attempts to retrieve those files while running the test suite off a `file://` URI.

Of course we can work around this limitation by accessing the test suite via HTTP (e.g. using [SimpleHTTPServer](http://www.linuxjournal.com/content/tech-tip-really-simple-http-server-python) or [servedir](https://github.com/remy/servedir)) - however, the need for this is not immediately obvious.

Thus I'd suggest that Blanket attempts to detect these errors and provides a simple explanation of what's going on.
